### ISSUE TITLE
Converted Whisper do not way to return timestamps

### ISSUE THREAD

**Issue by DimQ1** [microsoft/Olive#829]
**Describe the bug**
I converted the models to onnx format to work with float arrays. The model perfectly displays recognized text. But I can't get the timestamps for the recognized data.

**Other information**
 - OS:  Windows
 - Olive version: 0.5.0 
 - Onnxruntime-gpu: 1.16.3

**Comment by jambayk** [comment]
@DimQ1, thanks for bringing this up. The whisper beam search op has an option to add timestamp processing.
However, it does not work for `large-v3` model since there are some hardcoded values in the implementation that are not correct for it (https://github.com/microsoft/onnxruntime/blob/6d7519ede8298a422e84e70bfdf01cc46fbf76c3/onnxruntime/contrib_ops/cpu/transformers/logits_processor.h#L160)

For the supported models, I have implemented the required changes in this branch https://github.com/microsoft/Olive/tree/jambayk/whisper-logits/examples/whisper 

You can run the following to test it:
```bash
python prepare_whisper_configs.py --multilingual --predict_timestamps
python -m olive.workflows.run --config whisper_cpu_int8.json
python python test_transcription.py --config whisper_cpu_int8.json
```
`decoder_input_ids` only has 3 values now. Don't add the `no_timestamps` token.

The output comes out as something like:
```
"<|0.00|> The cut on his chest is still dripping blood.<|3.00|><|3.00|> The ache of his overstrained eyes,<|6.00|><|6.00|> even the soaring arena around him with thousands of spectators,<|10.00|><|10.00|> which revealed he's not worth thinking about.<|13.00|>"
```
So, you will have to do some extra post processing to get the offsets like in the pytorch implementation. The following works for me in python:
```python
import re

timestamp_pattern = r"<\|(\d+\.\d+)\|>"
float_pattern = r"(\d+\.\d+)"

offsets = []
current_text = None
current_start = None
for chunk in re.split(timestamp_pattern, text):
    # must be a text chunk
    if not re.match(float_pattern, chunk):
        current_text = chunk
        continue
    # must be a timestamp
    # start of a new chunk
    if current_start is None:
        current_start = float(chunk)
        continue
    # end of a chunk
    offsets.append({
        "text": current_text,
        "timestamp": (current_start, float(chunk))
    })
    # reset
    current_start = None
    current_text = None
```

Also, running a gpu model multiple times triggers a core dump on my machine. If the same happens to you, you can add `--skip_evaluation` when preparing the config.

**Comment by jambayk** [comment]
Please let me know if this works for you.

I also added transformers optimization pass to the `gpu_int8` workflow since it was initially skipped because of a bug in ort that was resolved long ago.

**Comment by DimQ1** [comment]
Hey. I found what the problem was. I tested this functionality on the whisper-large-v2 and I found out that in the my local file tokenizer.json haven't time stamp tokens. That's why it didn't work.
Thank you for example how extract timestamps. :)

**Comment by jambayk** [comment]
@DimQ1, did you do it using the branch I shared?

**Comment by DimQ1** [comment]
Yes. I used olive for the convert from main branch
